libs/dextool/source/dextool/compilation_db/package.d

```diff
@@ -714,16 +741,33 @@ ParseFlags parseFlag(CompileCommand cmd, const CompileCommandFilter flag_filter,
             skip,
             /// skip the next arg, if it is not a flag
             skipIfNotFlag,
+            /// use the next arg to create a complete path
+            checkingForEndQuotation,
         }
 
-        import std.path : buildNormalizedPath, absolutePath;
-        import std.array : appender, array;
+        import std.array : Appender, appender, array, join;
         import std.range : ElementType;
 
         auto st = State.keep;
         auto rval = appender!(string[]);
         auto includes = appender!(string[]);
         auto compiler = Compiler(r.length == 0 ? null : r[0]);
+        string path;
+        const string EMPTY_STRING;
+
+        string removeBackslashesAndQuotes(string arg) {
+            import std.string : replace;
+
+            return arg.replace(`\"`, EMPTY_STRING).replace(`"`, EMPTY_STRING);
```

Comment:
We have to assume that this may contain complex utf-8 strings because the underlying filesystem for do so.
This complicates the handling of the strings as they then have to be handled with care.
A utf-8 string contains code units, code points and grapheme. A grapheme is made up of one or more code points, and a code point is made up of one or more code units.
An iteration over a char[] as-is will iterate over the code points. There are no guarantee that a code point actually is a representable character. A code unit is just a piece of a character except for those rare cases where the input is ASCII.
A code point though is actually something representable. Same with a grapheme which can for example be a modifier for a character such as the dots above o in a ö. This fast becomes complicated.

Thus how to solve this conundrum?.
```d
return arg.byGrapheme.filter!(a => !a.among(Grapheme('\\'), Grapheme('"')).byCodePoint.text
```